Updated API, added tests, docs, export_html method and generate_html method

Prakriti-nith opened the pull request:
Right now, data is converted in the form of html table. I have shifted it to javascript array for the further server-side implementation (as datatable can take Array of Array). I will execute the server-side implementation (to load large set of data piece by piece) in this PR too. Parallel changes will be done accordingly in daru-datable in a separate PR.
I am also trying to resolve this [issue](https://github.com/Shekharrajak/daru-data_tables/issues/19). In this PR data is set correctly in the javascript array but the column names are overlapping for multiple tables.

Prakriti-nith commented:
I have updated the implementation to load a large set of data [here](https://github.com/Shekharrajak/daru-data_tables/pull/20). As the changes are in parallel in both the PRs, some of the tests will not pass.

Shekharrajak commented:
> I haven't added tests for generate_html as generate_html and export_html_file will then have the redundant tests. I am adding the tests for to_html in daru-data_tables.

We should  have testcases calling datatables adaptor here as well.

Prakriti-nith commented:
@Shekharrajak  I have added the test cases in this PR also.

Prakriti-nith commented:
In this PR, I have updated the API, added docs and tests and added `export_html` and `generate_html` methods.
Below are the screenshots of the examples:
![ddt1](https://user-images.githubusercontent.com/19255749/43045196-9620cb84-8dd1-11e8-8455-0eee1104f232.png)
![ddt2](https://user-images.githubusercontent.com/19255749/43045197-9aff7cc2-8dd1-11e8-9536-c1cf2514ea44.png)
